### PR TITLE
Blog layout and cards

### PR DESCRIPTION
**Pull request by FranekMarciniak**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

---

I agree to the terms of the RunBotics Contributor License Agreement.

### CODE REVIEW

**Comment by FranekMarciniak** [comment]
image
image
image

**Comment by FranekMarciniak** [comment]
TO BE DONE - BADGES

**Comment by FranekMarciniak** [comment]
image

File: runbotics/runbotics-orchestrator-ui/src/src-landing/components/BlogLayout/BlogLayout.module.scss [hunk 1]
```diff
@@ -0,0 +1,11 @@
+@import '#src-landing/scss/main.scss';
+
+.root {
+    composes: root from '#src-landing/components/SectionContent/SectionContent.module.scss';
+    display: flex;
+    flex-direction: row;
+    flex-wrap: wrap;
+    justify-content: center;
+    margin-top: 59px;
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/components/BlogLayout/BlogLayout.module.scss]: image

File: runbotics/runbotics-orchestrator-ui/src/src-landing/components/CardBadge/CardBadge.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import styles from './CardBadge.module.scss';
+
+interface Props {
+    text: string;
+    backgroundColor?: string;
+    textColor?: string;
+    className?: string;
+}
+const DEFAULT_BACKGROUND_COLOR = '#1E6BF0';
+const DEFAULT_TEXT_COLOR = '#FFF';
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/components/CardBadge/CardBadge.tsx:13]: move it to the global file

File: runbotics/runbotics-orchestrator-ui/src/src-landing/components/FeaturedBlogCard/FeaturedBlogCard.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+                    </Typography>
+                    <Typography variant="body4" className={styles.category}>
+                        {post.categories.items[0].title}
+                    </Typography>
+                </div>
+                <Typography variant="h3" className={styles.title}>
+                    {post.title}
+                </Typography>
+                <Typography variant="body3" className={styles.description}>
+                    {cutText(post.summary, 230)}
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/components/FeaturedBlogCard/FeaturedBlogCard.tsx:44]: text-overflow: ellipsis won't be sufficient?

Comment (preludium) [reply runbotics/runbotics-orchestrator-ui/src/src-landing/components/FeaturedBlogCard/FeaturedBlogCard.tsx:44]: even if strings have the same length their width can differ due to letter width

File: runbotics/runbotics-orchestrator-ui/src/src-landing/components/Navbar/Navbar.tsx [hunk 4]
```diff
@@ -36,7 +35,7 @@ const Navbar: FC<NavbarProps> = ({
             <ul className={`${styles.navLinkWrapper} ${isMobileVisible ? styles.minimized : navMobileStyle}`}>
                 <li className={styles.listItem}>
                     <Link
-                        href={`#${BENEFITS_SECTION_ID}`}
+                        href={`/#${BENEFITS_SECTION_ID}`}
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/components/Navbar/Navbar.tsx:38]: why this change?

File: runbotics/runbotics-orchestrator-ui/src/src-landing/views/sections/blog/BreadcrumbsSection/BreadcrumbsSection.module.scss [hunk 5]
```diff
@@ -0,0 +1,4 @@
+.root {
+    width: 100%;
+    height: 59px;
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/views/sections/blog/BreadcrumbsSection/BreadcrumbsSection.module.scss:3]: same

File: runbotics/runbotics-orchestrator-ui/src/src-landing/views/sections/blog/CardsSection/CardsSection.module.scss [hunk 6]
```diff
@@ -0,0 +1,36 @@
+@import '#src-landing/scss/main.scss';
+
+.root {
+    width: calc(100% - 270px);
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/views/sections/blog/CardsSection/CardsSection.module.scss]: what is 270px? seems magic

File: runbotics/runbotics-orchestrator-ui/next.config.js [hunk 7]
```diff
@@ -35,4 +35,8 @@ module.exports = {
         mailUsername: process.env.MAIL_USERNAME,
         mailPassword: process.env.MAIL_PASSWORD,
     },
+
+    images: {
+        domains: ["images.ctfassets.net"]
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/next.config.js:40]: this is the source for which images?

**Review by preludium — CHANGES_REQUESTED** [review]

File: runbotics/runbotics-orchestrator-ui/src/src-landing/views/sections/blog/CardsSection/CardsSection.module.scss (hunk 6 above)
Comment (FranekMarciniak) [reply runbotics/runbotics-orchestrator-ui/src/src-landing/views/sections/blog/CardsSection/CardsSection.module.scss]: width of filters section

File: runbotics/runbotics-orchestrator-ui/src/src-landing/views/sections/blog/BreadcrumbsSection/BreadcrumbsSection.module.scss (hunk 5 above)
Comment (FranekMarciniak) [reply runbotics/runbotics-orchestrator-ui/src/src-landing/views/sections/blog/BreadcrumbsSection/BreadcrumbsSection.module.scss:3]: height of nav

File: runbotics/runbotics-orchestrator-ui/src/src-landing/components/Navbar/Navbar.tsx (hunk 4 above)
Comment (FranekMarciniak) [reply runbotics/runbotics-orchestrator-ui/src/src-landing/components/Navbar/Navbar.tsx:38]: bc we want to use nav on blog page, if we do it without /, it would look like this 'url.com/blog#BENEFITS_SECTION_ID' and not 'url.com/#BENEFITS_SECTION_ID'

File: runbotics/runbotics-orchestrator-ui/src/src-landing/views/sections/blog/BreadcrumbsSection/BreadcrumbsSection.module.scss (hunk 5 above)
Comment (FranekMarciniak) [reply runbotics/runbotics-orchestrator-ui/src/src-landing/views/sections/blog/BreadcrumbsSection/BreadcrumbsSection.module.scss:3]: idk, temp value to be changed by someone doing this section

File: runbotics/runbotics-orchestrator-ui/src/src-landing/components/CardBadge/CardBadge.tsx (hunk 2 above)
Comment (FranekMarciniak) [reply runbotics/runbotics-orchestrator-ui/src/src-landing/components/CardBadge/CardBadge.tsx:13]: u sure? this is specific to badges

File: runbotics/runbotics-orchestrator-ui/src/src-landing/components/BlogCard/BlogCard.module.scss [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+$BADGE-LEFT: 20px;
+
+.root {
+    position: relative;
+    width: 100%;
+    padding: $CARD-PADDING;
+}
+
+.img {
+    position: relative !important;
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/components/BlogCard/BlogCard.module.scss:26]: ouch.

File: runbotics/runbotics-orchestrator-ui/src/src-landing/components/CardBadge/CardBadge.tsx [hunk 9]
```diff
@@ -0,0 +1,29 @@
+import React, { FC } from 'react';
+
+import Typography from '../Typography';
+import styles from './CardBadge.module.scss';
+
+interface Props {
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/components/CardBadge/CardBadge.tsx]: rename to CardBadgeProps

File: runbotics/runbotics-orchestrator-ui/src/src-landing/views/sections/blog/FiltersSection/FiltersSection.module.scss [hunk 10]
```diff
@@ -0,0 +1,13 @@
+@import '#src-landing/scss/main.scss';
+
+.root {
+    display: flex;
+    flex-direction: column;
+    width: 270px;
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/views/sections/blog/FiltersSection/FiltersSection.module.scss]: elevate to a constant variable

File: runbotics/runbotics-orchestrator-ui/next.config.js (hunk 7 above)
Comment (FranekMarciniak) [reply runbotics/runbotics-orchestrator-ui/next.config.js:40]: contentful cdn

File: runbotics/runbotics-orchestrator-ui/src/src-landing/components/BlogCard/BlogCard.module.scss (hunk 8 above)
Comment (FranekMarciniak) [reply runbotics/runbotics-orchestrator-ui/src/src-landing/components/BlogCard/BlogCard.module.scss:26]: i kno

File: runbotics/runbotics-orchestrator-ui/src/src-landing/components/Navbar/Navbar.tsx (hunk 4 above)
Comment (preludium) [reply runbotics/runbotics-orchestrator-ui/src/src-landing/components/Navbar/Navbar.tsx:38]: understandable

File: runbotics/runbotics-orchestrator-ui/src/src-landing/components/FeaturedBlogCard/FeaturedBlogCard.tsx [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+export const cutText = (text: string, length: number) => {
+    if (text.length < length) {
+        return text;
+    }
+    const cut = text.substring(0, length);
+    const lastSpace = cut.lastIndexOf(' ');
+    return cut.substring(0, lastSpace) + '...';
+};
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/components/FeaturedBlogCard/FeaturedBlogCard.tsx]: same implementation in `BlogCard` + imo let's use trim in if guard

File: runbotics/runbotics-orchestrator-ui/src/src-landing/views/sections/blog/CardsSection/CardsSection.tsx [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+
+import styles from './CardsSection.module.scss';
+
+interface BlogCardsSectionProps {
+    posts: BlogPost[];
+}
+
+const BlogCardsSection: FC<BlogCardsSectionProps> = ({ posts }) => (
+    <div className={styles.root}>
+        <div className={styles.topFilters}>top filters</div>
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-landing/views/sections/blog/CardsSection/CardsSection.tsx:15]: translation

**Review by preludium — CHANGES_REQUESTED** [review]

**Review by preludium — DISMISSED** [review]

**Review by preludium — APPROVED** [review]
